src/components/core/badge/badge.tsx

```diff
@@ -134,32 +171,48 @@ const BadgeContainer = styled('span', {
     },
 });
 
+const IconContainer = styled('div', {
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+});
+
 export const Badge = forwardRef<HTMLSpanElement, BadgeProps>(
     (
         {
-            icon_src,
-            icon_class,
+            prefix_icon,
+            suffix_icon,
             children,
             size = 'small',
             spacing = 'loose',
             visiblity = 'icon-and-label',
             label = 'regular',
-            iconAlt = 'Badge Icon',
         },
         ref,
     ) => {
-        const has_icon = visiblity === 'icon-and-label' || visiblity === 'icon-only';
-        const has_label = visiblity === 'icon-and-label' || visiblity === 'label-only';
+        const has_prefix_icon =
+            visiblity === 'icon-and-label' || visiblity === 'icon-only' || visiblity === 'icon-and-label-and-icon';
```

Comment:
```suggestion
        const has_prefix_icon =
            visiblity === 'icon-and-label' || visiblity === 'icon-only' || visiblity === 'icon-and-label-and-icon';
```
Refactor this to use Array.Some() as below